core/ops.rs

```diff
@@ -99,6 +99,16 @@ pub fn serialize_op_result<R: Serialize + 'static>(
   }))
 }
 
+/// Provides a convenient means for ops to assert if a buffer arg is passed
+pub fn assert_opbuf(
```

Comment:
I don't like functions named `assert` that don't panic. Maybe `opbuf_or_err`?

Reply:
Agree - also I don't think this belongs in core.

Reply:
> Agree - also I don't think this belongs in core.
> maybe `unwrap_or_type_error`

Where else would you put it ?

It's closely related to ops and potentially consumed by "all" ops ?

Reply:
The rusty naming might be ok_or_type_error /bikeshed

https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else

Reply:
> The rusty naming might be ok_or_type_error /bikeshed
> 
> https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else

Given that it's not a method, would that make it `opbuf_ok_or_type_error` ?
(Sounds a bit longwinded)

Part of the reason I used `assert` is that some of these actually were asserts before and some threw type errors

Reply:
> It's closely related to ops and potentially consumed by "all" ops.

I suppose core is fine...

Reply:
> > It's closely related to ops and potentially consumed by "all" ops.
> 
> I suppose core is fine...

I'm genuinely open to other suggestions, but given that it's shared/used by `op_crates`, `runtime`, etc... there doesn't really appear a better place to put it.

Reply:
@ry @hayd @lucacasonato I've replaced `assert_opbuf(...)?` with `.ok_or_else(null_opbuf)?` where `null_opbuf` comes from `deno_core::error`